src/Connection.js

```diff
@@ -104,11 +104,15 @@ class Connection extends EventEmitter {
     }
 
     send(controlLayerRequest) {
-        try {
-            this.socket.send(controlLayerRequest.serialize())
-        } catch (err) {
-            this.emit('error', err)
-        }
+        return new Promise((resolve, reject) => {
+            try {
+                this.socket.send(controlLayerRequest.serialize())
```

Comment:
Should you be hooking resolve to callback? Otherwise it will resolve immediately unless `send` is sync?

E.g. from network
```js
ws.send(message, (err) => {
                            if (err) {
                                reject(err)
                            } else {
                                debug('sent to %s message "%s"', recipientAddress, message)
                                resolve()
                            }
                        })
```

Not sure if browser websocket send supports callback though.

Reply:
> browser websocket send supports callback though.

it doesn't

Reply:
> unless send is sync

browser's `send` just enqueues the message to be sent, not sure there's a built-in way to detect that a specific message was received or failed to send

Reply:
This try/catch should perhaps just go specifically around the `controlLayerRequest.serialize()` part? Also not immediately sure why this is wrapped in a `Promise` since yeah, everything in here is sync

Reply:
True, removed Promise from send function